### PR TITLE
 Add pIC50 uncertainties in schema objects

### PR DESCRIPTION
**Pull request by kaminow**
Updated the CDD parsing functions to calculate and store the variance in the pIC50 measurement.

Replicate of previous PR #11

### CODE REVIEW

File: covid_moonshot_ml/datasets/moonshot.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 BASE_URL = "https://app.collaborativedrug.com/api/v1/vaults/5549/"
 ## All molecules with SMILES (public)
-ALL_SMI_SEARCH = "searches/8975987-kmJ-vR0fhkdccPw5UdWiIA"
+ALL_SMI_SEARCH = "searches/9469227-zd2doWwzJ63bZYaI_vkjXg"
```
Comment (mikemhenry) [inline covid_moonshot_ml/datasets/moonshot.py:3]: Why did the saved search change?

File: covid_moonshot_ml/datasets/utils.py [hunk 2]
```diff
@@ -136,6 +139,33 @@ def cdd_to_schema(cdd_csv, out_json, achiral=False):
...
+    ci_lower_key = (
+        "ProteaseAssay_Fluorescence_Dose-Response_Weizmann: IC50 "
+        "CI (Lower) (µM)"
+    )
+    ci_upper_key = (
+        "ProteaseAssay_Fluorescence_Dose-Response_Weizmann: IC50 "
+        "CI (Upper) (µM)"
+    )
+    ## Calculate 95% CI in pIC50 units based on IC50 vals (not sure if the
+    ##  difference should be taken before or after taking the -log10)
```
Comment (mikemhenry) [inline covid_moonshot_ml/datasets/utils.py:153]: @jchodera do you know the answer to this?

File: covid_moonshot_ml/datasets/utils.py [hunk 3]
```diff
@@ -159,14 +190,27 @@ def cdd_to_schema(cdd_csv, out_json, achiral=False):
         )
     compounds = ExperimentalCompoundDataUpdate(compounds=compounds)
 
-    with open(out_json, "w") as fp:
-        fp.write(compounds.json())
-    print(f"Wrote {out_json}", flush=True)
+    if out_json is not None:
```
Comment (mikemhenry) [inline covid_moonshot_ml/datasets/utils.py:193]:
```suggestion
    if out_json:
```

File: covid_moonshot_ml/datasets/utils.py [hunk 4]
```diff
@@ -159,14 +190,27 @@ def cdd_to_schema(cdd_csv, out_json, achiral=False):
...
+        with open(out_json, "w") as fp:
+            fp.write(compounds.json())
+        print(f"Wrote {out_json}", flush=True)
+    if out_csv is not None:
```
Comment (mikemhenry) [inline covid_moonshot_ml/datasets/utils.py:197]:
```suggestion
    if out_csv:
```

File: covid_moonshot_ml/datasets/utils.py [hunk 5]
```diff
@@ -247,9 +321,23 @@ def cdd_to_schema_pair(cdd_csv, out_json):
 
     ep_list = EnantiomerPairList(pairs=enant_pairs)
 
-    with open(out_json, "w") as fp:
-        fp.write(ep_list.json())
-    print(f"Wrote {out_json}", flush=True)
+    if out_json is not None:
```
Comment (mikemhenry) [inline covid_moonshot_ml/datasets/utils.py:324]:
```suggestion
    if out_json:
```

File: covid_moonshot_ml/datasets/utils.py [hunk 6]
```diff
@@ -247,9 +321,23 @@ def cdd_to_schema_pair(cdd_csv, out_json):
...
+        with open(out_json, "w") as fp:
+            fp.write(ep_list.json())
+        print(f"Wrote {out_json}", flush=True)
+    if out_csv is not None:
```
Comment (mikemhenry) [inline covid_moonshot_ml/datasets/utils.py:328]:
```suggestion
    if out_csv:
```

**Review by mikemhenry — CHANGES_REQUESTED** [review]

File: covid_moonshot_ml/datasets/moonshot.py (hunk 1 above)
Comment (kaminow) [reply covid_moonshot_ml/datasets/moonshot.py:3]: I made a new saved search on CDD that contains the upper/lower CI in the export, so updated to the URL for the new search

File: covid_moonshot_ml/datasets/utils.py (hunk 2 above)
Comment (apayne97) [reply covid_moonshot_ml/datasets/utils.py:153]: `-log(x+y) != -log(x) + -log(y)`
`-log(x) + -log(y) = -log(x*y)`

Therefore you should add the +/- confidence intervals to the calculated value *before* taking the log.

I think that's right?

Comment (kaminow) [reply covid_moonshot_ml/datasets/utils.py:153]: sorry for the confusion, I think I meant to delete this comment at some point. I believe this is answered here:
https://choderalab.slack.com/archives/C03B535P36C/p1657215011957369

Comment (jchodera) [reply covid_moonshot_ml/datasets/utils.py:153]: That's right---you want to convert each IC50 separately, then use that pIC50 confidence interval as desired.

**Review by apayne97 — APPROVED** [review]

**Comment by jchodera** [comment]
> The error in the log scale will be at least slightly asymmetric (if the original errors were symmetric).

Yes, this is why this whole strategy is imperfect. Ultimately, we'll want to work with confidence intervals when possible, but arsenic/cinnibar doesn't support this right now.

**Review by mikemhenry — APPROVED** [review]

**Pull request merged**